docs/RFCS/20220711_external_connections.md

```diff
@@ -0,0 +1,407 @@
+operations such as backup, restore, changefeed, import, and export.
+
+#### GRANT
+
+A user or role requires the GRANT privilege to in turn grant privileges to other users. The owner of
+an object implicitly has the `GRANT OPTION` for all privileges. If a privilege is
+granted `WITH GRANT OPTION` to a user, then the user can in turn grant that privilege to other
+users.
+
+To support granting the above privileges we will add the following statements to the
+existing `GRANT` SQL grammar:
+
+```sql
+GRANT ALL ON EXTERNAL CONNECTION <name> TO <role_spec_list> [WITH GRANT OPTION];
+
+GRANT USAGE, GRANT ON EXTERNAL CONNECTION <name> TO <role_spec_list> [WITH GRANT OPTION];
+```
+
+An External Connection is a non-descriptor backed object and will store a synthetic privilege
+descriptor in the `system.privileges` table. The privileges will be addressable by the unique path
+`/externalconnection/<external-connection-name>` and will be marshaled and unmarshaled using the
+APIs provided by `pkg/sql/syntheticprivilege`.
+
+As mentioned above, the default owner of an External Connection is the user that created the object.
+The owner of the object can alter the External Connection to have a new owner if the user executing
+the command is a member of the new owner role. This will be done using:
+
+```sql
+ALTER EXTERNAL CONNECTION <name> OWNER TO <role_spec_list>;
+```
+
+### Observability
+
+Variations of the `SHOW` statement will be used to provide observability into the External
+Connection objects in a cluster. Similar to other implementations, the show statements will be
+powered by a delegate that issues `SELECT` queries against the
+underlying `system.external_resources` table. The permissions required to run particular `SHOW`
+queries are discussed below:
+
+```sql
+SHOW GRANTS ON EXTERNAL CONNECTION <name>;
+```
+
+No privileges are required for a user to view the grants on an External Connection. The query will
+output a table with each row specifying the `name` of the External Connection, the `grantee`,
+the `privilege_type`, and whether the privilege `is_grantable`. `is_grantable` will be true if the
+privilege was granted `WITH GRANT OPTION` to the user or role.
+
+```sql
+SHOW EXTERNAL CONNECTION <name>;
+
+SHOW EXTERNAL CONNECTIONS;
+```
+
+The above queries will output a table with each row specifying the `name` of the External
+Connection, the `created` timestamp, the `owner`, and the `connection_type` of the object. There
+will be an additional column for `connection_details` whose content will be powered by
+the `DisplayString` method exposed by the `ConnectionDetails` interface. This query can be run by
+any user or role regardless of their privileges since it will only display redacted representations
+of the external connections.
+
+```sql
+SHOW CREATE EXTERNAL CONNECTION <name>;
+```
+
+This query can only be run by the root, admin, or the owner of the External Connection object. It
```

Comment:
any admin?

I mean, that makes sense as any admin can just read the system table, but just want to be clear

Reply:
correct, any admin. I'll pick that change up when I push next.